src/java/org/apache/cassandra/utils/SystemInfo.java

```diff
@@ -0,0 +1,214 @@
+     * @return The amount of virtual memory allowed to be allocatedby the current process/user.
+     */
+    long getVirtualMemoryMax() {
+        return si.getOperatingSystem().getCurrentProcess().getVirtualSize();
+    }
+
+    /**
+     *
+     * @return The amount of swap space allocated on the system.
+     */
+    long getSwapSize() {
+        return si.getHardware().getMemory().getVirtualMemory().getSwapTotal();
+    }
+
+    /**
+     * @return the PID of the current system.
+     */
+    public long getPid() {
+        return si.getOperatingSystem().getProcessId();
+    }
+
+    /**
+     * Checks if a value if valide (i.e. value >= min or value == INFINITY..
+     * @param value the value to check.
+     * @param min the minimum value.
+     * @return true if value is valid.
+     */
+    private boolean valid(long value, long min)
+    {
+        return value >= min || value == INFINITY;
+    }
+
+    /**
+     * Tests if the system is running in degraded mode.
+     * If the system is running in degraded mode this method will log information.
+     * @return @{code true} if the system is in degraded mode, @{code false} otherwise.
+     */
+    public boolean warnIfRunningInDegradedMode()
+    {
+        boolean degraded = false;
+        StringBuilder sb = new StringBuilder();
+
+        if (oshi.SystemInfo.getCurrentPlatform() == PlatformEnum.LINUX)
+        {
+            // only check proc on nproc linux
+            if (!valid(getMaxProcess(), EXPECTED_NPROC))
+            {
+                sb.append("Number of processes should be >= ").append(EXPECTED_NPROC).append("\n");
+                degraded = true;
+            }
+        } else
+        {
+            degraded = true;
+            sb.append("System is running ").append(platform()).append(", Linux OS is recommended");
+        }
+
+        if (getSwapSize() > 0)
+        {
+            sb.append("Swap should be disabled\n");
+            degraded = true;
+        }
+        if (!valid(getMaxProcess(), EXPECTED_NPROC))
+        {
+            sb.append("Minimum value for max number of processes should be >= ").append(EXPECTED_NPROC).append("\n");
+            degraded = true;
+        }
+        if (!valid(getVirtualMemoryMax(), EXPECTED_AS))
+        {
+            sb.append("Amount of available address space should be >= ").append(EXPECTED_AS).append("\n");
+            degraded = true;
+        }
+        if (!valid(getMaxOpenFiles(), EXPECTED_MIN_NOFILE))
+        {
+            sb.append("Minimum value for max open files should be >=  ").append(EXPECTED_MIN_NOFILE).append("\n");
+            degraded = true;
+        }
+        if (degraded) {
+            if (CassandraRelevantProperties.TEST_IGNORE_SIGAR.getBoolean())
+            {
+                logger.warn("Cassandra server running in degraded mode.\n" + sb.toString());
```

Comment:
I am not sure all these individual appends via `sb` should ended on `\n`. Why can it not be just a simple sentence with a dot? Why new lines? A simple line is just fine.